File: CSharp/DP.UnitTest/UtFactory.cs
using System;
using System.Diagnostics;
using DP.Domain.Samples.Strategy;
using DP.Domain.Samples.Facade;
using DP.UnitTest.Utility;
using Xunit;
using Xunit.Abstractions;
using DP.Domain.Samples.Bridge;
using DP.Domain.Samples.Factory;

namespace DP.UnitTest
{
    public class UtFactory
    {
        private readonly ITestOutputHelper _output;

        public UtFactory(ITestOutputHelper output)
        {
            this._output = output;
            System.Diagnostics.Trace.Listeners.Add(new XunitTraceListener(this._output));
        }

       
        [Fact]
        public void TestStcDbFactory()
        {
            #region Create Dbcontext for DataMart
            var dmDbcontext =  StcDbFactory.Create(DbEnum.DataMart);
            dmDbcontext.Connect();
            Assert.Equal(DbEnum.DataMart.ToString(), dmDbcontext.Server);
            #endregion

            #region Create Dbcontext for Online
            var olDbcontext =  StcDbFactory.CreateOnline();
            Assert.Equal(DbEnum.Online.ToString(), olDbcontext.Server);
            olDbcontext.Connect();
            #endregion
        }

        [Fact]
        public void TestAbsDbFactory()
        {
            #region Create Dbcontext for DataMart
            var dmFactory = new DataMartDbFactory();
            var dmDbcontext =  dmFactory.Create();
            dmDbcontext.Connect();
            Assert.Equal(DbEnum.DataMart.ToString(), dmDbcontext.Server);
            #endregion

            #region Create Dbcontext for History
            var hsFactory = new HistoryDbFactory();
            var hsDbcontext =  hsFactory.Create();
            hsDbcontext.Connect();
            Assert.Equal(DbEnum.History.ToString(), hsDbcontext.Server);
            #endregion
        }

        [Fact]
        public void TestGenericDbFactory()
        {
            #region Create Dbcontext for DataMart
            var dmDbcontext =  GenericDbFactory<DataMartDbContext>.Create();
            dmDbcontext.Connect();
            Assert.Equal(DbEnum.DataMart.ToString(), dmDbcontext.Server);
            #endregion

            #region Create Dbcontext for Online
            var olDbcontext =  GenericDbFactory<OnlineDbContext>.Create();
            olDbcontext.Connect();
            Assert.Equal(DbEnum.Online.ToString(), olDbcontext.Server);
            #endregion
        }
    }
}

File: Python/Samples/Factory/UtFactory.py
import unittest
from Models import DbContext, DbEnum
from AbsDbFactory import AbsDbFactory, DataMartDbFactory, HistoryDbFactory, OnlineDbFactory
from StcDbFactory import StcDbFactory


class UtFactory(unittest.TestCase):

    def test_static_factory(self):
        dmDbcontext = StcDbFactory.create(DbEnum.DataMart)
        dmDbcontext.connect()
        self.assertEqual(dmDbcontext.server, DbEnum.DataMart.name)

        olDbcontext = StcDbFactory.createOnline()
        olDbcontext.connect()
        self.assertEqual(olDbcontext.server, DbEnum.Online.name)

    def test_abstract_factory(self):

        dmFactory = DataMartDbFactory()
        dmDbcontext =  dmFactory.create()
        dmDbcontext.connect();
        self.assertEqual(dmDbcontext.server, DbEnum.DataMart.name)


if __name__ == '__main__':
    unittest.main()
